lib/std/os.zig

```diff
@@ -5091,6 +5091,10 @@ pub const RealPathError = error{
 /// extra `/` characters in `pathname`.
 /// The return value is a slice of `out_buffer`, but not necessarily from the beginning.
 /// See also `realpathZ` and `realpathW`.
+/// Note: All /-separated components of the supplied path must resolve to a
+/// directory, or a symlink to a directory, except for the last component
+/// Use `fs.path.resolve` or `fs.path.relative` to get dir of file (`file/..`).
```

Comment:
`realpath` should fully resolve any path containing relative `.` and `..` components on any POSIX-compliant system as far as I remember. It is only Windows that is the problem where the NT kernel call we issue is unable to deal with those which implies it is up to the user to `resolve` the relative components first.

Reply:
The non-working semantics (`pathtofolder/file/..`) of `fs.path.relative` can be seen here https://github.com/matu3ba/chepa/commit/f34084cbf103ca9bbf75db94bf4b1835a8c9b6aa#diff-d924ca21d81d7d5a59eb9e10d3e2689c4c58a7e28e6ecce6a064701666f5a730L173

> It is only Windows that is the problem where the NT kernel call we issue is unable to deal with those which implies it is up to the user to resolve the relative components first.

Could we on Windows work around the problem with calling relative on the subpath and then call resolve for each extending subpath? If you can point me to the things I need for testing, I can play abit to see if that could work.
Kind related is https://github.com/ziglang/zig/issues/4812.